staging/src/k8s.io/apimachinery/pkg/runtime/serializer/cbor/direct/direct.go

```diff
@@ -24,10 +24,16 @@ import (
 )
 
 func Marshal(src interface{}) ([]byte, error) {
+	if err := modes.RejectCustomMarshalers(src); err != nil {
+		return nil, err
+	}
 	return modes.Encode.Marshal(src)
 }
 
 func Unmarshal(src []byte, dst interface{}) error {
+	if err := modes.RejectCustomMarshalers(dst); err != nil {
+		return err
+	}
 	return modes.Decode.Unmarshal(src, dst)
 }
```

Comment:
The description of this PR was useful to me, can we somehow include that knowledge into the code?